imdraw/imdraw.go

```diff
@@ -128,7 +128,9 @@ func (imd *IMDraw) Draw(t pixel.Target) {
 // Push adds some points to the IM queue. All Pushed points will have the same properties except for
 // the position.
 func (imd *IMDraw) Push(pts ...pixel.Vec) {
-	imd.Color = pixel.ToRGBA(imd.Color)
+	if _, ok := imd.Color.(pixel.RGBA); !ok {
+		imd.Color = pixel.ToRGBA(imd.Color)
+	}
```

Comment:
it can't be. The problem here is that pixel.ToRGBA returns a pixel.RGBA, which *implements*, but *isn't*, a color.Color interface. So the compiler has to box the result, which is an object allocation, which is why this is so unexpectedly effective.